toolchain/sem_ir/inst_namer.cpp

```diff
@@ -631,12 +631,13 @@ auto InstNamer::CollectNamesInBlock(ScopeId top_scope_id,
             sem_ir_->facet_types().Get(inst.facet_type_id);
         bool has_where = facet_type_info.other_requirements ||
                          !facet_type_info.rewrite_constraints.empty();
-        if (auto interface = facet_type_info.TryAsSingleInterface()) {
-          const auto& interface_info =
-              sem_ir_->interfaces().Get(interface->interface_id);
+        if (facet_type_info.extend_constraints.size() == 1) {
+          const auto& interface_info = sem_ir_->interfaces().Get(
+              facet_type_info.extend_constraints.front().interface_id);
           add_inst_name_id(interface_info.name_id,
                            has_where ? "_where.type" : ".type");
-        } else if (facet_type_info.impls_constraints.empty()) {
+        } else if (facet_type_info.extend_constraints.empty() &&
+                   facet_type_info.self_impls_constraints.empty()) {
           add_inst_name(has_where ? "type_where" : "type");
```

Comment:
Can this `has_where = true` case still exist? The only way here is `type where _something_` but there's nothing you can do on the RHS other than `.Self impls type` which doesn't end up with an interface constraint?